projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,316 @@
+    private Date currentDate;
+    // Indicates whether user has sent 'haser'.
+    private boolean sentHaser = true;
+    // Indicates whether user has sent more then 5 lawsuits already in the past year.
+    private boolean sentMoreThanFiveLawsuits = false;
+    private String moreThanFiveLawsuits ="הגיש";
+    private String lessThanFiveLawsuits = "לא הגיש/ו";
+    private String fiveLawsuitsStatus =""; // Init onCreate().
+    private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    private String claimCaseSubscription = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    // Cases: Didn't subscribe / Sent 'haser' but got spam.
+    private String claimCase = ""; //  Init on fillInTemplate().
+
+    // User:
+    private String userPrivateName = "שם פרטי";
+    private String userLastname = "שם משפחה";
+    private String userID = "ת ז משתמש";
+    private String userAddress = "כתובת משתמש";
+    private String userPhone = "טלפון משתמש";
+    private String userFax = "פקס משתמש";
+
+    // First spam company's details:
+    private String companyName = "שם החברה";
+    private String companyId = "מספר ח.פ";
+    private String companyAddress = "כתובת החברה";
+    private String companyPhone = "טלפון החברה";
+    private String companyFax = "פקס חברה";
+
+    // Second spam company's det ails:
+    private String company2Name = "שם החברה 2";
+    private String company2Id = "מספר ח.פ 2";
+    private String company2Address = "כתובת החברה 2";
+    private String company2Phone = "טלפון החברה 2";
+    private String company2Fax = "פקס חברה 2";
+// ----------------------------- - -------------------------------------------------------:
+    Button createPdfButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_lawsuit_pdf);
+        createPdfButton = (Button) findViewById(R.id.button_createPDF);
+
+        // Check Read/Write permissions:
+        isWriteStoragePermissionGranted();
+
+        // Initialize folders needed for activity:
+        initDirStructure();
+
+        // Read lawsuit template's content:
+        templateContent = readTemplate();
+
+        // Init dates:
+        receivedSpamDate = new Date();
+        currentDate = new Date();
+        // Update date values:
+        dateAndTimeFormatter = new SimpleDateFormat(DATE_TIME_FORMAT);
+        dateAndTimeFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+        dateFormatter = new SimpleDateFormat(DATE_FORMAT);
+        dateFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+
+        // Generate lawsuit PDF OnClick:
+        createPdfButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    createPdf();
+                } catch (IOException e) {
```

Comment:
I think you can put this error inside `createPdf()`.
I believe you can then simplify this whole call using lambdas (Android Studio might suggest it as a lint warning).

Reply:
Moved it to createPdf().
About the lambdas, do we need to add Java 8 configurations?
https://developer.android.com/studio/write/java8-support